private/rules/maven_install.bzl

```diff
@@ -96,37 +109,45 @@ def maven_install(
     # @unpinned_maven//:pin / Coursier to update maven_install.json, and bazel build
     # //... immediately after with the updated artifacts.
 
-    coursier_fetch(
-        # Name this repository "unpinned_{name}" if the user specified a
-        # maven_install.json file. The actual @{name} repository will be
-        # created from the maven_install.json file in the coursier_fetch
-        # invocation after this.
-        name = name if maven_install_json == None else "unpinned_" + name,
-        repositories = repositories_json_strings,
-        artifacts = artifacts_json_strings,
-        fail_on_missing_checksum = fail_on_missing_checksum,
-        fetch_sources = fetch_sources,
-        fetch_javadoc = fetch_javadoc,
-        excluded_artifacts = excluded_artifacts_json_strings,
-        generate_compat_repositories = generate_compat_repositories,
-        version_conflict_policy = version_conflict_policy,
-        override_targets = override_targets,
-        strict_visibility = strict_visibility,
-        strict_visibility_value = strict_visibility_value,
-        maven_install_json = maven_install_json,
-        resolve_timeout = resolve_timeout,
-        use_credentials_from_home_netrc_file = use_credentials_from_home_netrc_file,
-        use_starlark_android_rules = use_starlark_android_rules,
-        aar_import_bzl_label = aar_import_bzl_label,
-        duplicate_version_warning = duplicate_version_warning,
-    )
+    if resolver != "coursier":
```

Comment:
nit: mind if we reverse this if statement so we have `if resolver == "courser":  courser_fetch` and have `generate_pin_repository` in the else clause